helix-core/src/syntax.rs

```diff
@@ -40,13 +40,14 @@ where
 }
 
 #[derive(Debug, Serialize, Deserialize)]
+#[serde(deny_unknown_fields)]
```

Comment:
I don't think it is a good idea to deny unknown fields, if later when we add or remove a new field it might be a breaking change that affects the user.

Reply:
The default `languages.toml` is embedded in the binary, meaning that if new fields are added to it, they won't end up reaching the user unless they get an updated binary.

Reply:
Rationale for adding it in the first place: https://github.com/helix-editor/helix/pull/982#discussion_r743418384